### PR TITLE
Add Session TTL with Remember me feature

### PR DESCRIPTION
**Pull request by ottaviano**
image

### CODE REVIEW

File: src/Security/AdherentLoginTimestampRecorder.php [hunk 1]
```diff
@@ -28,8 +28,8 @@ public function onSecurityInteractiveLogin(InteractiveLoginEvent $event)
             return;
         }
 
-        if (!$token instanceof UsernamePasswordToken) {
-            throw new \RuntimeException(sprintf('Authentication token must be a %s instance.', UsernamePasswordToken::class));
+        if (!$token instanceof TokenInterface) {
```
Comment (B-Galati) [inline src/Security/AdherentLoginTimestampRecorder.php]: This test is basically useless now as `$event` is always returning a `TokenInterface`.
Let's remove it no ?

**Comment by B-Galati** [comment]
@ottaviano Perhaps the checkbox should be checked by default

File: templates/security/adherent_login.html.twig [hunk 2]
```diff
@@ -27,7 +27,12 @@
 
                     <a href="{{ path('forgot_password') }}" class="form__forgotten-pw">Oublié ?</a>
                 </li>
-
+                <li>
+                    <div class="b__nudge--top text--small form__checkbox form__checkbox--large" id="field-conditions">
+                        <input type="checkbox" id="remember_me" name="_remember_me"/>
+                        <label class="form form__label" for="remember_me">Se souvenir de  moi</label>
```
Comment (thomasbisignani) [inline templates/security/adherent_login.html.twig]: There is a useless whitespace in the text

File: templates/security/adherent_login.html.twig [hunk 3]
```diff
@@ -27,7 +27,12 @@
 
                     <a href="{{ path('forgot_password') }}" class="form__forgotten-pw">Oublié ?</a>
                 </li>
-
+                <li>
+                    <div class="b__nudge--top text--small form__checkbox form__checkbox--large" id="field-conditions">
```
Comment (thomasbisignani) [inline templates/security/adherent_login.html.twig]: The `field-conditions` id is really needed here ?

File: app/migrations/Version20180321121625.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+use Doctrine\DBAL\Schema\Schema;
+
+class Version20180321121625 extends AbstractMigration
+{
+    public function up(Schema $schema)
+    {
+        $this->addSql('CREATE TABLE `rememberme_token` (
+            `series` char(88) UNIQUE PRIMARY KEY NOT NULL,
+            `value` char(88) NOT NULL,
+            `lastUsed` datetime NOT NULL,
```
Comment (hhamon) [inline app/migrations/Version20180321121625.php]: `last_used` no?

File: templates/security/adherent_login.html.twig (hunk 2 above)
Comment (ottaviano) [reply templates/security/adherent_login.html.twig]: fixed, thank yoy @thomasbisignani :D

File: templates/security/adherent_login.html.twig (hunk 3 above)
Comment (ottaviano) [reply templates/security/adherent_login.html.twig]: nop, removed 👍

File: app/migrations/Version20180321121625.php (hunk 4 above)
Comment (ottaviano) [reply app/migrations/Version20180321121625.php]: no, we should keep it in camelCase, it is used here: https://github.com/symfony/symfony/blob/master/src/Symfony/Bridge/Doctrine/Security/RememberMe/DoctrineTokenProvider.php#L84

**Review by illusionOfParadise — APPROVED** [review]

**Pull request merged**